force-app/main/default/lwc/rd2UpdateNextPayment/rd2UpdateNextPayment.js

```diff
@@ -0,0 +1,327 @@
+                    
+                    break;
+                }
+            }
+        }
+    }
+
+    /***
+     * @description Sets installment datatable columns
+     */
+    handleColumns(response) {
+        if (response && response.dataTable) {
+            let tempColumns = response.dataTable.columns;
+            this.columns = [];
+
+            for (let i = 0; i < tempColumns.length; i++) {
+                if (tempColumns[i].fieldName !== "pauseStatus") {
+                    this.columns.push(tempColumns[i]);
+                }
+            }
+        }
+    }
+
+    /***
+    * @description Handles button display. In the case of permission, field visibility
+    * or RD closed error, [OK] button is displayed and [Save] button is not displayed.
+    */
+    handleButtonsDisplay() {
+        this.isSaveDisplayed = !this.isLoading && this.permissions.hasAccess && !this.permissions.isBlocked;
+
+        // Disable data display and Save button when installments are not returned
+        if (this.installments == null && this.isSaveDisplayed) {
+            this.isSaveDisplayed = false;
+            this.hasAccess = false;
+        }
+    }
+
+    /***
+    * @description Save Next Payment Amount: Updates Schedule with Next Payment Amount
+    */
+    handleSave() {
+        this.clearError();
+        this.isLoading = true;
+        try {
+            handleNextPaymentAmount({
+                rdId: this.recordId,
+                nextPaymentAmount: this.nextPaymentAmount
+            })
+                .then((jsonResponse) => {
+                    const response = isNull(jsonResponse) ? null : JSON.parse(jsonResponse);
+                    const isSuccess =
+                        isNull(response) ||
+                        response.statusCode === HTTP_CODES.Created ||
+                        response.statusCode === HTTP_CODES.OK;
+                    if (isSuccess) {
+                        this.handleSaveSuccess();
+                    } else {
+                        const message = this.rd2Service.getCommitmentError(response);
+                        this.handleError(message);
+                    }
+                })
+                .catch((error) => {
+                    this.handleError(error);
+                });
+        } catch (error) {
+            this.handleError(error);
+        }
+    }
+
+    /***
+    * @description Displays message that the save has been successful
+    * and closes the modal.
+    */
+    handleSaveSuccess() {
+        const message = this.labels.changeNextInstallmentSuccess;
+        showToast(message, '', 'success', []);
+
+        this.dispatchEvent( new CloseActionScreenEvent() );
+    }
+
+    /***
+    * @description Records the latest Next Payment Amount value
+    * and checks if the [Save] button should be enabled.
+    */
+    handleNextPaymentAmountChange(event) {
+        this.nextPaymentAmount = event.detail.value;
```

Comment:
Also just a non-blocking comment here, this function could be broken down so its easy for us to figure out what it does.